### PR TITLE
Port to C++98

### PR DESCRIPTION
**Pull request by ericliu201208**

### CODE REVIEW

File: cpp/QrCodeGeneratorDemo.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/* 
+﻿/*
```
Comment (mathias-gopp-lgs) [inline cpp/QrCodeGeneratorDemo.cpp]: what is here the change?

File: cpp/QrCodeGeneratorDemo.cpp [hunk 2]
```diff
@@ -57,35 +54,34 @@ int main() {
 	return EXIT_SUCCESS;
 }
 
-
```
Comment (mathias-gopp-lgs) [inline cpp/QrCodeGeneratorDemo.cpp]: no benefit of removing lines

File: cpp/QrCodeGeneratorDemo.cpp [hunk 3]
```diff
@@ -34,11 +33,9 @@
 #include <vector>
 #include "qrcodegen.hpp"
 
-using std::uint8_t;
 using qrcodegen::QrCode;
 using qrcodegen::QrSegment;
 
-
```
Comment (mathias-gopp-lgs) [inline cpp/QrCodeGeneratorDemo.cpp]: no benefit of removing lines

File: cpp/QrCodeGeneratorDemo.cpp [hunk 4]
```diff
@@ -57,35 +54,34 @@ int main() {
...
 static void doBasicDemo() {
 	const char *text = "Hello, world!";              // User-supplied text
-	const QrCode::Ecc errCorLvl = QrCode::Ecc::LOW;  // Error correction level
+	const QrCode::Ecc errCorLvl = QrCode::LOW;  // Error correction level
 	
 	// Make and print the QR Code symbol
 	const QrCode qr = QrCode::encodeText(text, errCorLvl);
 	printQr(qr);
 	std::cout << toSvgString(qr, 4) << std::endl;
+
```
Comment (mathias-gopp-lgs) [inline cpp/QrCodeGeneratorDemo.cpp]: no benefit of adding a line

File: cpp/qrcodegen.cpp [hunk 5]
```diff
@@ -355,7 +355,9 @@ QrCode::QrCode(int ver, Ecc ecl, const vector<uint8_t> &dataCodewords, int msk)
 	drawFormatBits(msk);  // Overwrite old format bits
 	
 	isFunction.clear();
-	isFunction.shrink_to_fit();
+	//isFunction.shrink_to_fit();
+
```
Comment (mathias-gopp-lgs) [inline cpp/qrcodegen.cpp]: what is the purpose of this commented functions? Can they be removed?

File: cpp/qrcodegen.cpp [hunk 6]
```diff
@@ -820,11 +822,13 @@ BitBuffer::BitBuffer()
...
 
-void BitBuffer::appendBits(std::uint32_t val, int len) {
+void BitBuffer::appendBits(uint32_t val, int len) {
 	if (len < 0 || len > 31 || val >> len != 0)
 		throw std::domain_error("Value out of range");
 	for (int i = len - 1; i >= 0; i--)  // Append bit by bit
 		this->push_back(((val >> i) & 1) != 0);
 }
 
+
```
Comment (mathias-gopp-lgs) [inline cpp/qrcodegen.cpp]: no benefit of adding lines

**Review by mathias-gopp-lgs — CHANGES_REQUESTED** [review]
There are certain line addings and removals which are not needed.
There are out commented functions where it is not clear whether they are needed.

File: cpp/QrCodeGeneratorDemo.cpp (hunk 1 above)
Comment (jingsheng-huang) [reply cpp/QrCodeGeneratorDemo.cpp]: Updated

File: cpp/QrCodeGeneratorDemo.cpp (hunk 2 above)
Comment (jingsheng-huang) [reply cpp/QrCodeGeneratorDemo.cpp]: Updated

File: cpp/QrCodeGeneratorDemo.cpp (hunk 3 above)
Comment (jingsheng-huang) [reply cpp/QrCodeGeneratorDemo.cpp]: Updated

File: cpp/QrCodeGeneratorDemo.cpp (hunk 4 above)
Comment (jingsheng-huang) [reply cpp/QrCodeGeneratorDemo.cpp]: Updated

File: cpp/qrcodegen.cpp (hunk 5 above)
Comment (jingsheng-huang) [reply cpp/qrcodegen.cpp]: shrink_to_fit() is for C++ 11, to remove the extra capacity inside a vector. 
Replaced by swap() for C++98, same purpose.

File: cpp/qrcodegen.cpp (hunk 6 above)
Comment (jingsheng-huang) [reply cpp/qrcodegen.cpp]: Updated

**Review by mathias-gopp-lgs — APPROVED** [review]

**Pull request merged**